manager/volume.go

```diff
@@ -613,12 +616,48 @@ func (m *VolumeManager) Expand(volumeName string, size int64) (v *longhorn.Volum
 		return nil, fmt.Errorf("cannot expand volume before replica scheduling success")
 	}
 
-	if v.Spec.Size > size {
-		return nil, fmt.Errorf("cannot expand volume %v with current size %v to a smaller size %v", v.Name, v.Spec.Size, size)
+	kubernetesStatus := &v.Status.KubernetesStatus
```

Comment:
Add rounding here size = util.RoundUpSize(size)
This rounds the PVC assigned size up to align with the longhorn blockvolume size.
It's a nice to have but not a requirement.

Below before assigning to the v.spec.Size you do need to roundUp, see below comment.

Reply:
changed